### ISSUE TITLE
Data Exported from Siemens WinCC SCADA in Table format and viewed in UAExpert client doesnot match with data fetched with node_id.read_raw_history(starttime,endtime,0)

### ISSUE THREAD

**Issue by rajnishnationfirst** [FreeOpcUa/python-opcua#1091]
Hi Everyone ,

I fetched a **archived** tag data  using **node_id.read_raw_history(starttime,endtime,0)** it feches 122 records having 
 0.5 second (or  **500 Microsecond**) and it is fecthed successfully .

<img width="934" alt="read_raw_history_snap" src="https://user-images.githubusercontent.com/64892368/86530444-5c5b7d80-bed6-11ea-8705-049d67b26897.PNG">

**Also when i viewed in a table data ( feature of WinCC Siemens SCADA ) i get the data as below format** :

<img width="942" alt="PDF_view" src="https://user-images.githubusercontent.com/64892368/86530486-9cbafb80-bed6-11ea-8316-7b1a61dd92a5.PNG">

**At the same i viewd the data in UA Expert and i viewed the data as attched below** 👍 
<img width="703" alt="UA_Expert" src="https://user-images.githubusercontent.com/64892368/86530491-b78d7000-bed6-11ea-829c-a4600f514481.PNG">

Issues i am facing : 

The data in UA Expert and the TABLE VIEW ( Feature of SIEMENS WInCC SCADA ) is matching , while the data recieved via read_raw_history doesnot matched ( which some matches as seen in the below attachment .

<img width="946" alt="Value_matched" src="https://user-images.githubusercontent.com/64892368/86530521-f4596700-bed6-11ea-98b1-b860024e1e07.PNG">

What may be the issue in **node_id.read_raw_history(starttime,endtime,0)**?

How can i get the correct data using **node_id.read_raw_history(starttime,endtime,0)** or i will have to use some othr functions like **node_id.history_read(params)**

It would be great if anyone can assist me on this and provide me the appropriate reply which will help me to proceed further.

Thanks and Regards
Rajnish Vishwakarma

**Comment by brubbel** [comment]
It seems to me that the timestamps are matching (if you shift the first red box 1 row up).
However, the last value at second 48.578 (pdf row 10) is not matching.

**Comment by rajnishnationfirst** [comment]
Thanks brubbel for your kind reply.
I have now changed the range of UA Expert and snapped it for you to view .
Now you can pls check that the UA Expert & Table VIEW ( feature of SIEMENS WInCC SCADA ) is same and the data rcived snapped using read_raw_history is different ( I mean to say , that UA Expert and Table view has same data and equal but wwhen matched with node_id.read_raw_history(starttime,endtime,0 ) does not matched.

**Pls have a look to all 3 snaps :**
**UA Expert View ( range corrected )**

<img width="958" alt="UA_Expert_New" src="https://user-images.githubusercontent.com/64892368/86531042-9aa76b80-bedb-11ea-9753-c5806cb1eb62.PNG">

PDF View and Data View : 

<img width="946" alt="Value_matched" src="https://user-images.githubusercontent.com/64892368/86531054-b3b01c80-bedb-11ea-8467-f9e6e8cb5138.PNG">

Also lastly response snap from node_id.read_raw_history(starttime,endtime,0)

<img width="934" alt="read_raw_history_snap" src="https://user-images.githubusercontent.com/64892368/86531080-e6f2ab80-bedb-11ea-8af8-71c85bb1e57a.PNG">

Yes the red block marked matches with UAExpert & Table View Data ( from SIEMENS WInCC SCADA )

Also if you can help me with some questions 👍 

1) As some data matches (in Red Block )using data fetched using **node_id.read_raw_history(starttime,endtime,0)** , does its a server side issue of issue in **node_id.read_raw_history(starttime,endtime,0)** ?

2) If there is any issue in **node_id.read_raw_history(starttime,endtime,0)** , what is the correct method / function to read and fetched data ?

Thanks and Regards
Rajnish Vishwakarma

**Comment by zerox1212** [comment]
From what I remember there is a time conversion in Python for the binary protocol. Native `DateTime` objects are probably converted and maybe some precision is lost. Might be worth digging into what timestamp is coming across the wire and comparing it to what ends up in python DateTime object.

**Comment by rajnishnationfirst** [comment]
Thanks Andrew or your reply:
I am quering as given below for a datetime object 

**Code:**
```
start_time = datetime.datetime(2020, 7, 6, 10, 35)
end_time = datetime.datetime(2020, 7, 6,  10,41)

node .read_raw_history(start_time, end_time, 0))
```


Is the above date time object correct?
If wrong , how can i configure datetime object as per your reply you gave above.

Andrew , pls assist me i am stuck here due to this.

Thanks 
Rajnish Vishwakarma

**Comment by brubbel** [comment]
I still do not understand why you consistently highlight the column off-by-one?
![image](https://user-images.githubusercontent.com/5667690/86905594-95803180-c112-11ea-9140-8bce10df854d.png)

**Comment by zerox1212** [comment]
I do not think it has to do with your start and end times. It has to due with time conversion that happens here: https://github.com/FreeOpcUa/python-opcua/blob/13b4e1249e06a3d3feef09afd04139eb774207bc/opcua/ua/ua_binary.py#L45

You can investigate this for loss of time precision.
